testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeLdap.java

```diff
@@ -16,25 +16,52 @@
 import com.google.common.collect.ImmutableList;
 import com.google.inject.Inject;
 import io.trino.tests.product.launcher.docker.DockerFiles;
-import io.trino.tests.product.launcher.env.EnvironmentConfig;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
 import io.trino.tests.product.launcher.env.common.Hadoop;
+import io.trino.tests.product.launcher.env.common.OpenLdap;
 import io.trino.tests.product.launcher.env.common.Standard;
 import io.trino.tests.product.launcher.env.common.TestsEnvironment;
 import io.trino.tests.product.launcher.testcontainers.PortBinder;
 
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TEMPTO_PROFILE_CONFIG;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_CONFIG_PROPERTIES;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
 @TestsEnvironment
 public class EnvSinglenodeLdap
-        extends AbstractEnvSinglenodeLdap
+        extends EnvironmentProvider
 {
+    private final DockerFiles dockerFiles;
+    private final PortBinder portBinder;
+
     @Inject
-    public EnvSinglenodeLdap(Standard standard, Hadoop hadoop, DockerFiles dockerFiles, PortBinder portBinder, EnvironmentConfig config)
+    public EnvSinglenodeLdap(Standard standard, Hadoop hadoop, OpenLdap openLdap, DockerFiles dockerFiles, PortBinder portBinder)
     {
-        super(ImmutableList.of(standard, hadoop), dockerFiles, portBinder, config);
+        super(ImmutableList.of(standard, hadoop, openLdap));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
     }
 
     @Override
-    protected String getPasswordAuthenticatorConfigPath()
+    public void extendEnvironment(Environment.Builder builder)
     {
-        return "conf/environment/singlenode-ldap/password-authenticator.properties";
+        builder.addPasswordAuthenticator("ldap", forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-ldap/password-authenticator.properties")));
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer.withCopyFileToContainer(
+                    forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-ldap/config.properties")),
+                    CONTAINER_TRINO_CONFIG_PROPERTIES);
+
+            portBinder.exposePort(dockerContainer, 8443);
```

Comment:
Isn't this already exposed? I think it is something that should be exposed for coordinator as it is a common case. If not maybe it should land in `Standard` instead?

Reply:
It depends on the type of Coordinator - Only with for server with TLS 8443 is exposed.

Reply:
Standard exposes 8080 - But 8443 is special port so we could expose them on Environment specific.

Reply:
Maybe we should have `Tls` env extender?